### DISCUSSION TITLE
How to keep formulas in sync with upstream releases? (python)

### DISCUSSION THREAD

**Discussion by ssbarnea** [Homebrew/discussions#1237] [Everyday usage]
I am long term happy homebrew user and happen to maintain several more or less popular python based command line tools (pip installable).

While for me there is no brainer to use `pip3` or `pipx` install python based tools and apps, I do also wish to make it easier for those that may not happen to master the python packaging ecosystem hidden "gems".

While writing a formula for a specific version of a python application is not really a big deal (if you do not have too many dependencies), I see the process almost impossible to scale for applications that do make releases often.

If an app is released only once or twice a year, no problem, you likely have time to manually refresh that formula. Still, if an app makes releases like 20-40 times a year, you are doomed, nobody will  have the nerves (time) to update those dependencies after each release, a very boring task.

Is there a solution for addressing this issue?  Ideally I would like to see a solution where this can be integrated in the normal release pipelines of a project, so when they publish a new wheel on pypi, the cask should also be updated.

Did anyone reached such a level of integration or not really?

**Comment by carlocab** [comment]
For formulae, you can try using https://github.com/dawidd6/action-homebrew-bump-formula. Not sure if there's an equivalent for casks.
